File: lua/hamming/hamming.lua
local hamming = {}

function hamming.compute(a, b)
  local sum = 0

  -- I'm assuming a and b are the same length
  for i = 1, #a do
    if a:byte(i) ~= b:byte(i) then
      sum = sum + 1
    end
  end

  return sum
end

return hamming

File: c/hamming/src/hamming.c
int compute(char* a, char* b) {
  int difference = 0;

  char x, y;
  int i = 0;

  x = *(a + i);
  y = *(b + i);

  while (!(x == '\0' || y == '\0')) {
    if (x != y) difference++;

    i++;
    x = *(a + i);
    y = *(b + i);
  }

  return difference;
}
